### PR TITLE
feat: add pr review icons (approved, requested changes, commented, dismissed)

### LINKED ISSUE #682: In the Gitify menu bar (Mac) tray it has some indication that you "approved", "requested changes", or "commented" on the PR
### 📝 Provide a description of the new feature

# Where
In the notifications tray from Gitify menu bar icon. For each tray item it would be nice to have some indication if I approved, requested changes, or just commented on it.

# Problem space
I often will approve a PR and a conversation will continue between other people (this can be very noisy). Some pull requests I want to engage and others I want to unsubscribe. However, for the life of me, I cannot recall or identify pull requests by the title and cannot remember if I reviewed it or not.

When I click on a tray item it removes it out of the tray (as expected) but then it is gone and I can't click unsubscribe through Gitify's tray item (I suppose I could through GitHubs notifications but I like Gitify).

# A suggested solution might be

Have an icon/badge for the tray item that indicates if I approved it already. Then I can choose to open or unsubscribe.

The "request-changes" comes up less frequently but is an interesting use case as it would help me prioritize which ones to review since if I'm blocking a pull request I would likely want to review those changes faster to unblock the person. Not really an issue as the person could dismiss my review.

### ➕ Additional Information

_No response_

### PR DESCRIPTION
**Pull request by setchy**
Closes #682

For Pull Request notifications, when `detailed notifications setting is enabled, fetch PR reviews and find the most recent for the authenticated user

Will show an additional icon on the notification row, as pictured below.  

Screenshot 2024-05-01 at 3 18 05 PM

**Feedback Areas**
I'd appreciate feedback on the following:
- are there better positions for the icon?
- mouse-over text (closely modeled off the GitHub UI wording)

**Broader discussion**
This has also got me thinking more generally if we could/should show PR Approval icons for all reviewers.  If so, what would we show when there is multiple people with feedback?

### CODE REVIEW

**Comment by setchy** [comment]
@bhingston-va - as the feature requestor - would really value your input on this 🙏

**Comment by afonsojramos** [comment]
How does it look on hover? When there's more icons?

**Comment by bhingston-va** [comment]
Looks great to me! This is awesome. I'll be honest this was more of need when the notifications would duplicate/inconsistent state. This inconsistent state was fixed in v5.4.0 I believe. That said this would be really cool. Thanks for actioning on it so fast. 🎉

**Comment by bhingston-va** [comment]
>How does it look on hover? When there's more icons?

Looking again. On hover, I see the other icons likely collide or get squished. I wonder if putting the icon with the "latest update". With the idea it is just our review and could be later iterated on to support multiple reviewers by having the icon of the reviewer.

before|after
---|---
Screenshot 2024-05-02 at 8 52 01 AM|Screenshot 2024-05-02 at 8 52 01 AM

**Comment by bhingston-va** [comment]
>Show up to 3 icons per notification footer (approved, changes requested, commented/dismissed)

Minor detail, consider always including "my review" if it exists.

**Comment by setchy** [comment]
Following the above discussion, I've pushed updates that now
* add the icons to the notification row footer
* fetch the latest review for each reviewer
* generate appropriate descriptions

Screenshot 2024-05-03 at 6 27 57 AM
Screenshot 2024-05-03 at 6 28 06 AM

**Comment by setchy** [comment]
Here is an example of multiple review states using stubbed data

Screenshot 2024-05-03 at 6 53 26 AM

File: src/utils/subject.ts [hunk 1]
```diff
@@ -245,9 +257,67 @@ async function getGitifySubjectForPullRequest(
...
+        users: [prReview.user.login],
+      });
+      continue;
+    }
+
+    reviewerFound.users.push(prReview.user.login);
+  }
+
+  // Sort the reviews by state for consistent draw order
+  return [
```
Comment (setchy) [inline src/utils/subject.ts:312]: I'll update this soon (once back at a computer) to a much simpler alpha sort

File: src/utils/subject.ts [hunk 2]
```diff
@@ -15,15 +18,22 @@ import {
...
+import {
+  fetchDiscussion,
+  getLatestDiscussionComment,
+  getTokenForHost,
+} from './helpers';
 
 export async function getGitifySubjectDetails(
   notification: Notification,
-  token: string,
+  accounts: AuthState,
```
Comment (afonsojramos) [inline src/utils/subject.ts:32]: Why did we move the `getTokenForHost` call to here? Not that it matters much, just trying to understand the reasoning

Comment (setchy) [reply src/utils/subject.ts]: Reverted @afonsojramos - thanks for picking this up

**Review by afonsojramos — APPROVED** [review]
Nice job, looking good!

**Comment by setchy** [comment]
@bhingston-va - does this address your feature request suggestion?

**Pull request merged**